packages/cozy-client/src/models/konnectorFolder.js

```diff
@@ -0,0 +1,216 @@
+
+const FILES_DOCTYPE = 'io.cozy.files'
+const PERMISSIONS_DOCTYPE = 'io.cozy.permissions'
+
+export const ensureKonnectorFolder = async (client, { konnector, account }) => {
+  const permissions = client.collection(PERMISSIONS_DOCTYPE)
+  const files = client.collection(FILES_DOCTYPE)
+  const adminFolder = await folderModel.ensureMagicFolder(
+    client,
+    folderModel.MAGIC_FOLDERS.ADMINISTRATIVE,
+    '/Administratif'
+  )
+  const path = buildFolderPath(konnector, account, {
+    // @ts-ignore
+    administrative: adminFolder.path
+  })
+  const folder =
+    (await statDirectoryByPath(client, path)) ||
+    (await createDirectoryByPath(client, path))
+
+  await permissions.add(konnector, buildFolderPermission(folder))
+  await files.addReferencesTo(konnector, [folder])
+
+  return folder
+}
+
+/**
+ * Creates a directory from a given path
+ *
+ * @param  {Object}  client CozyClient
+ * @param  {string}  path   Directory path
+ * @returns {Promise<import('../types').FileDocument>}         Directory attributes
+ */
+export const createDirectoryByPath = async (client, path) => {
+  const { data } = await client
+    .collection(FILES_DOCTYPE)
+    .createDirectoryByPath(path)
+  return data
+}
+
+/**
+ * Retrieves a directory from its path
+ *
+ * @param  {Object}  client CozyClient
+ * @param  {string}  path   Directory path
+ * @returns {Promise<import('../types').FileDocument>}        Created io.cozy.files document
+ */
+export const statDirectoryByPath = async (client, path) => {
+  try {
+    const response = await client.collection(FILES_DOCTYPE).statByPath(path)
+    if (response.data.trashed) {
+      return null
+    }
+    return response.data
+  } catch (error) {
+    if (error && error.status === 404) return null
+    throw new Error(error.message)
+  }
+}
+
+/**
+ * Build folder path for a given konnector and a given account.
+ *
+ * If konnector.folders[0].defaultDir exists, it is used as default directory.
+ *
+ * Occurrences of following strings in base directory are replaced by:
+ * * `$administrative`: Administrative folder
+ * * `$photos`: Photos folder
+ *
+ * Occurrences of following strings in path are replaced by:
+ * * `$account: Account label (id or name)`
+ * * `$konnector`: Konnector name
+ *
+ * If no konnectors.folders[0].defaultDir is set, the default dir used is
+ * *  `$administrative/$konnector/$account`
+ *
+ * @param  {Object} konnector Konnector document
+ * @param  {Object} account   Account document
+ * @param  {Object} folders   Object containing a mapping from folder
+ * identifiers (ex: $administrative) to their localized values (ex:
+ * Administratif).
+ * @returns {String}           The result path
+ */
+
+export const buildFolderPath = (konnector, account, folders = {}) => {
+  const fullPath = get(
+    konnector,
+    // For now konnectors are only defining one folder in their folders array
+    'folders[0].defaultDir',
+    '$administrative/$konnector/$account'
+  )
```

Comment:
Do we really need lodash get? I guess `konnector?.folders?.[0]?.defaultDir || '$administrative/$konnector/$account'` would do the job?

Reply:
Also, trim could be easily replaced https://youmightnotneed.com/lodash ;).

Reply:
A bit less trivial, though ;)